backend/src/main/modules/helper.py

```diff
@@ -0,0 +1,1010 @@
+import shutil
+from xlutils.copy import copy
+import shutil
+import re
+from xlrd import open_workbook
+from xlutils.copy import copy as xl_copy
+import logging.handlers
+import time
+from logging.handlers import TimedRotatingFileHandler
+import xlsxwriter
+import argparse
+import sys
+from os import path
+import openpyxl
+from openpyxl import Workbook, load_workbook
+from openpyxl.styles import Color, PatternFill, Font, Border
+from openpyxl.styles import colors
+from openpyxl.cell import Cell
+from common_config import *
+import threading
+
+millisecond = None
+childsolutionid = ""
+regex = "\"?([-a-zA-Z0-9.`?{}]+@\w+\.\w+)\"?"
+class Helpers:
+    def __init__(self):
+        self.millisecond = None
+
+    def programCreation(accessToken, parentFolder, externalId, pName, orgIds,creatorKeyCloakId, creatorName):
+        global programIdForSuccess, programExternalId
+       
+        programCreationurl = internal_kong_ip_core +  programcreationurl
+        
+        programExternalId = externalId
+        payload = json.dumps({
+            "externalId": externalId,
+            "name": pName,
+            "description": "Program",
+            "resourceType": [
+                "program"
+            ],
+            "language": [
+                "English"
+            ],
+            "keywords": [],
+            "concepts": [],
+            "createdFor": orgIds,
+            "rootOrganisations": orgIds,
+            "startDate": startDateOfProgram,
+            "endDate": endDateOfProgram,
+            "imageCompression": {
+                "quality": 10
+            },
+            "creator": creatorName,
+            "owner": creatorKeyCloakId,
+            "author": creatorKeyCloakId,
+            # "metaInformation": {
+            #     # "state":entitiesPGM.split(","),
+            #     # "role": mainRole.split(",")
+            #     },
+                "requestForPIIConsent":False
+            })
+
+        
+        headers = {'X-authenticated-user-token': accessToken,
+                   'internal-access-token': internal_access_token,
+                   'Content-Type': 'application/json',
+                   'Authorization':authorization}
+
+        # program creation 
+        responsePgmCreate = requests.request("POST", programCreationurl, headers=headers, data=(payload))
+        print(responsePgmCreate.text)
+
+        fileheader = [pName, ('Program Sheet Validation'), ('Passed')]
+        if responsePgmCreate.status_code == 200:
+            responsePgmCreateResp = responsePgmCreate.json()
+            programIdForSuccess = responsePgmCreateResp['result']["_id"]
+        else:
+            # terminate execution
+            print("Program creation API failed. Please check logs.")
+
+# Th    is function checks for the sequince
+
+    
+
+    def createFileStructForProgram(programFile):
+        print("programFile:-------------",programFile)
+        if not os.path.isdir('resourceFile'):
+            os.mkdir('resourceFile')
```

Comment:
Don't create folders in the working directory. fix a static location and create folders. 
Keep in mind that this is not a local script, it is a service. There can be multiple users at the same time. If you are creating folders with static name/path. It can lead to potential race condition.